### ISSUE TITLE
LUN Fuzzy vs. RegEx Search

### ISSUE THREAD

**Issue by singlecheeze** [Infinidat/infinibox-csi-driver#118]
https://github.com/Infinidat/infinibox-csi-driver/blob/24830ee357a56ec55b94f4c788ed5c601b28ab15/storage/fcnode.go#L780

I wrote a quick comparison of the `findFcDisk` [method ](https://github.com/Infinidat/infinibox-csi-driver/blob/24830ee357a56ec55b94f4c788ed5c601b28ab15/storage/fcnode.go#L780)and compared it to the [upstream](https://github.com/kubernetes/kubernetes/blob/948afe5ca072329a73c8e79ed5938717a5cb3d21/pkg/volume/fc/fc_util.go#L64). I provisioned 300 LUNs on my SAN, and am invoking a debug pod in OpenShift and outputting `ls /host/dev/disk/by-path/`.  

Note: The [upstream inclusion](https://github.com/kubernetes/kubernetes/commit/ddcc448070fcecb67d98b0a239607d9bb8fd3161) of the `$`  in the RegEx which is very important.

With `strings.Contains` in the current implementation, and given a `/host/dev/disk/by-path/ of fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-10` there are many instances where higher LUN numbers could be returned.

The current driver returns as soon as the first `strings.Contains` is matched, which is illustrated by (I deleted LUN 1 from the array and rescanned to make sure CoreOS picked it up):
```text
Testing contains for fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-1:
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-10
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-100
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-101
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-102
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-103
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-104
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-105
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-106
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-107
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-108
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-109
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-11
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-110
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-111
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-112
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-113
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-114
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-115
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-116
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-117
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-118
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-119
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-12
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-120
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-121
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-122
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-123
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-124
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-125
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-126
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-127
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-128
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-129
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-13
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-130
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-131
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-132
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-133
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-134
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-135
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-136
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-137
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-138
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-139
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-14
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-140
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-141
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-142
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-143
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-144
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-145
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-146
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-147
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-148
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-149
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-15
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-150
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-151
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-152
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-153
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-154
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-155
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-156
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-157
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-158
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-159
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-16
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-160
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-161
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-162
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-163
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-164
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-165
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-166
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-167
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-168
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-169
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-17
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-170
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-171
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-172
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-173
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-174
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-175
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-176
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-177
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-178
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-179
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-18
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-180
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-181
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-182
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-183
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-184
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-185
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-186
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-187
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-188
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-189
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-19
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-190
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-191
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-192
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-193
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-194
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-195
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-196
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-197
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-198
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-199

Testing REGEX expression ^(pci-.*-fc|fc)-0x100000109bc3914e-0x21000024ff1e85b6-lun-1$:

Testing contains for fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-11:
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-11
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-110
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-111
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-112
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-113
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-114
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-115
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-116
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-117
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-118
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-119

Testing REGEX expression ^(pci-.*-fc|fc)-0x100000109bc3914e-0x21000024ff1e85b6-lun-11$:
fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-11
```
`Testing contains for fc-0x100000109bc3914e-0x21000024ff1e85b6-lun-1` should not match, but it does and LUN 10 would be returned by the current driver, when in reality, this is a completely different device.

**Comment by jmccormick2001** [comment]
we are testing a solution to this, the solution is going to replace the strings.Contain() with a regex MatchString().  The regex pattern is taken from the upstream example:

FcPath := "^(pci-.*-fc|fc)-0x" + wwn + "-lun-" + lun + "$"

This will fix the matching logic to find the correct device path.    

Also along with this fix we are changing the polling interval and frequency for the findMultipathDevice() function.  The old values were to sleep 1 Second between polling iterations with up to 10 iterations.  With the upcoming change the polling sleep time is set to 500 Milliseconds with up to 20 iterations.  This should improve the performance of the findMultipathDevice().

Additionally, timing log messages at the DEBUG level have been added so that users can look at the node logs for FC timing values if needed.

**Comment by jmccormick2001** [comment]
the solution or fix to this will be made available via a patch release (2.16.1).  We are planning that release for hopefully next week.

**Comment by singlecheeze** [comment]
@jmccormick2001 I think I just found another issue in the same method. It appears that for LUN numbers > 255, `by-path` will start listing pci devices differently:

```
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-1
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-2
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-3
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-4
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-5
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-6
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-7
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-8
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-9
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-10
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-11
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-12
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-13
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-14
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-15
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-16
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-17
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-18
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-19
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-20
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-21
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-22
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-23
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-24
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-25
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-26
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-27
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-28
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-29
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-30
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-31
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-32
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-33
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-34
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-35
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-36
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-37
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-38
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-39
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-40
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-41
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-42
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-43
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-44
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-45
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-46
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-47
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-48
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-49
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-50
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-51
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-52
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-53
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-54
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-55
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-56
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-57
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-58
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-59
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-60
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-61
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-62
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-63
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-64
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-65
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-66
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-67
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-68
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-69
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-70
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-71
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-72
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-73
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-74
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-75
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-76
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-77
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-78
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-79
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-80
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-81
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-82
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-83
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-84
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-85
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-86
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-87
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-88
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-89
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-90
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-91
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-92
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-93
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-94
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-95
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-96
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-97
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-98
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-99
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-100
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-101
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-102
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-103
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-104
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-105
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-106
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-107
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-108
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-109
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-110
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-111
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-112
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-113
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-114
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-115
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-116
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-117
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-118
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-119
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-120
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-121
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-122
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-123
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-124
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-125
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-126
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-127
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-128
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-129
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-130
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-131
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-132
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-133
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-134
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-135
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-136
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-137
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-138
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-139
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-140
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-141
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-142
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-143
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-144
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-145
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-146
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-147
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-148
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-149
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-150
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-151
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-152
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-153
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-154
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-155
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-156
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-157
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-158
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-159
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-160
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-161
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-162
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-163
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-164
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-165
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-166
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-167
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-168
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-169
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-170
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-171
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-172
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-173
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-174
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-175
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-176
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-177
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-178
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-179
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-180
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-181
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-182
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-183
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-184
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-185
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-186
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-187
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-188
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-189
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-190
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-191
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-192
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-193
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-194
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-195
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-196
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-197
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-198
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-199
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-200
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-201
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-202
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-203
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-204
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-205
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-206
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-207
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-208
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-209
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-210
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-211
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-212
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-213
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-214
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-215
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-216
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-217
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-218
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-219
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-220
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-221
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-222
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-223
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-224
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-225
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-226
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-227
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-228
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-229
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-230
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-231
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-232
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-233
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-234
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-235
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-236
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-237
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-238
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-239
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-240
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-241
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-242
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-243
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-244
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-245
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-246
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-247
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-248
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-249
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-250
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-251
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-252
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-253
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-254
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-255
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-0x0100000000000000
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-0x0101000000000000
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-0x0102000000000000
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-0x0103000000000000
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-0x0104000000000000
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-0x0105000000000000
pci-0000:82:00.0-fc-0x21000024ff1e85b6-lun-0x0106000000000000
```

The REGEX does a search for the `pci` prefix with a REGEX OR operator `|`.

Seems this s known in the upstream and the general guidance is to not trust `by-path`, instead only use `by-id`:  
https://github.com/kubernetes/kubernetes/issues/48639  
https://github.com/kubernetes/kubernetes/issues/45024  
https://github.com/kubernetes/kubernetes/pull/48741  

It appears the only use of `by-path` is in two files in the CSI:  
https://github.com/search?q=repo%3AInfinidat%2Finfinibox-csi-driver%20by-path&type=code  

Looks like an optional method was added in upstream k8s for searching specifically by ID:  
https://github.com/kubernetes/kubernetes/blob/948afe5ca072329a73c8e79ed5938717a5cb3d21/pkg/volume/fc/fc_util.go#L84

**Comment by jmccormick2001** [comment]
@singlecheeze nice find, it will require a fix for sure.

**Comment by jmccormick2001** [comment]
@singlecheeze fyi, the 255 limit fix will land in the 2.17.0 release in the coming weeks, this fix is coded and it seems to test ok.

**Comment by singlecheeze** [comment]
@jmccormick2001 is that by chance pushed to github or will it be pushed when 4.17 is ready?

**Comment by ekaulberg** [comment]
@singlecheeze we generally do not push prerelease source to Github, but
we'll push it when 2.17.0 is ready. Let's correspond via email if there's a
need ahead of that.

**Comment by jmccormick2001** [comment]
this fix, using by-id instead of by-path, will be included in the 4.17.0 release which is about to happen.
